admin/services/check-complete.service.js

```diff
@@ -25,6 +28,18 @@ checkCompleteService.completeCheck = async function (completedCheck) {
   // TODO: This timestamp should be recorded in the application service tier instead
   completedCheck.receivedByServerAt = moment.utc()
 
+  const existingCheck = await checkDataService.sqlFindOneByCheckCode(completedCheck.data.pupil.checkCode)
+  if (!existingCheck.startedAt) {
```

Comment:
what if the existing check is null or undefined?
that will throw an exception

Reply:
It will throw `TypeError: Cannot read property 'startedAt'` - the controller will trap this and return a server error.